### PR TITLE
dutch auction for #26

### PR DESCRIPTION
**Pull request by brossetti1**
dutch auction for https://github.com/Geo-Web-Project/core-contracts/issues/26

### CODE REVIEW

File: contracts/FairLaunchClaimer.sol [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    function claim(
+        address user,
+        uint256 initialContributionRate,
+        bytes calldata claimData
+    ) external payable onlyRole(CLAIM_ROLE) {
+        require(block.timestamp < auctionStart, "auction has not started yet");
+        require(block.timestamp > auctionEnd, "geneisis is complete");
+        require(msg.value >= _requiredBid(), "bid is too low");
+
+        (uint64 baseCoordinate, uint256[] calldata path) = claimData;
```
Comment (codynhat) [inline contracts/FairLaunchClaimer.sol:59]: Try:

```solidity
(uint64 baseCoordinate, uint256[] path) = abi.decode(claimData, (uint64, uint256[]));
```

File: contracts/FairLaunchClaimer.sol [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    bytes32 public constant CLAIM_ROLE = keccak256("CLAIM_ROLE");
+
+    /// @notice License
+    ERC721License public license;
+
+    /// @notice Parcel
+    GeoWebParcel public parcel;
+
+    /// @notice Auction
+    AuctionSuperApp public auctionApp;
```
Comment (codynhat) [inline contracts/FairLaunchClaimer.sol:22]: Is this actually used anywhere?

File: contracts/FairLaunchClaimer.sol [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    uint256 public auctionStart;
+    /// @notice when the required bid amount reaches its minimum value.
+    uint256 public auctionEnd;
+    /// @notice start price of the genesis land auction. Decreases to endingBid between auctionStart and auctionEnd.
+    uint256 public startingBid;
+    /// @notice the final/minimum required bid reached and maintained at the end of the auction.
+    uint256 public endingBid;
+
+    /// @notice Emitted when a parcel is purchased
+    event ParcelPurchased(uint256 indexed parcelId, address indexed to);
```
Comment (codynhat) [inline contracts/FairLaunchClaimer.sol:34]: Should this event have "claim" in the name? Like `ParcelClaimed` or `ClaimedParcel`? Could be confused with someone purchasing an existing parcel from someone else.

File: tasks/FairLaunchAuction.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import { ethers } from "ethers";
+import { task, types } from "hardhat/config";
+
+task("deploy:fair-claimer", "Deploy FairLaunchClaimer").setAction(
+  async (args, hre) => {
+    const FairLaunchClaimer = await hre.ethers.getContractFactory("FairLaunchClaimer");
```
Comment (codynhat) [inline tasks/FairLaunchAuction.ts:6]: I started using the generated factories in the deploy scripts.

**Review by codynhat — CHANGES_REQUESTED** [review]

File: contracts/FairLaunchClaimer.sol (hunk 2 above)
Comment (brossetti1) [reply contracts/FairLaunchClaimer.sol:22]: not in this contract, it will be used in the reclaimer, i can remove it here

**Pull request merged**